larky/src/main/java/com/verygood/security/larky/modules/NetworkTokenModule.java

```diff
@@ -101,16 +110,32 @@ public Dict<String, Object> getNetworkToken(
       String currencyCode,
       String cryptogramType,
       String vgsMerchantId,
+      @Nullable Object transactionType,
       StarlarkThread thread)
       throws EvalException {
     if (pan.trim().isEmpty()) {
       throw Starlark.errorf("pan argument cannot be blank");
     }
     final Optional<NetworkTokenService.NetworkToken> networkTokenOptional;
     try {
-      networkTokenOptional =
-          networkTokenService.getNetworkToken(pan, cvv, amount, currencyCode, cryptogramType,
-              vgsMerchantId);
+      if (networkTokenService instanceof NetworkTokenServiceV2 v2Service) {
+        networkTokenOptional =
+            v2Service.getNetworkTokenV2(
+                NetworkTokenServiceV2.GetNetworkTokenRequest.builder()
+                    .panAlias(pan)
+                    .cvv(cvv)
+                    .amount(amount)
+                    .currencyCode(currencyCode)
+                    .cryptogramType(cryptogramType)
+                    .merchantId(vgsMerchantId)
+                    .transactionType(
+                        transactionType instanceof String ? (String) transactionType : null)
+                    .build());
+      } else {
+        networkTokenOptional =
```

Comment:
When would this be true?

Reply:
When the provided `networkTokenService` object is just a `NetworkTokenService` rather than `NetworkTokenServiceV2`. I have concerns regarding if we make a breaking change here in starlarky and make a new release, atlas-commons and atlas-horizon will get stuck with legacy version of starlarky. Unless, the new `NetworkTokenService` is implemented in atlas-commons immediately after this PR merged and release.

So breaking up `NetworkTokenService` and `NetworkTokenServiceV2`, making `NetworkTokenService` still workable is just to ensure even if we merge this and release new larky right now, it shouldn't break `atlast-commons` when upgrading larky without implementing with the new method signature.

If that's not a concern, I can just drop the two interfaces approach and change the `NetworkTokenService` interface directly. Please let me know what you think.

Reply:
Not sure I follow. You can update Larky, release it, update the parent pom, then apply the new parent pom to atlas-commons, release that (which updates PP automatically) then apply that new PP to horizon/everest

Reply:
The only concern I have is that if we merge this PR, and somehow cannot update `atlast-horizon` in time, other releases will be blocked until the another PR is done. But since as if this is not a concern and we are going to fix the stuff and merge it quickly, I am going to simplify the code by using just one interface with new signature then.

Reply:
fixed, now I only use one interface. I will have `atlast-commons` PR coming up soon